apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -0,0 +1,128 @@
+}
+
+const initialSelectModule = (searchParams: any, modulenames: string[]) => {
+    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
+
+    if (!!paramModule && modulenames.includes(paramModule)) {
+        return paramModule;
+    } else {
+        return modulenames[0];
+    }
+};
+
+function PkgModuleViewWrapper({ id, modules }: Props) {
+    const modulenames = modules.map(([name], idx) => name);
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [query, setQuery] = useState('');
+    const [selectedModule, setSelectedModule] = useState(
+        initialSelectModule(searchParams, modulenames)
+    );
+
+    const filteredModules =
+        query === ''
+            ? modulenames
+            : modules
+                  .filter(([name]) => name.startsWith(query))
+                  .map(([name]) => name);
+
+    useEffect(() => {
+        setSearchParams({ module: selectedModule });
+    }, [selectedModule, setSearchParams]);
+
+    const submitSearch = useCallback(() => {
+        setSelectedModule((prev: string) =>
+            filteredModules.length === 1 ? filteredModules[0] : prev
+        );
+    }, [filteredModules]);
+
+    return (
+        <div className="flex flex-wrap border-0 border-y border-solid border-sui-grey-45">
+            <div className="w-full md:w-[15vw] pr-[20px]">
+                <Combobox value={selectedModule} onChange={setSelectedModule}>
+                    <div className="box-border border border-sui-grey-50 border-solid rounded-[6px] h-[34px] shadow-sm placeholder-sui-grey-65 pl-3 w-full flex my-[10px] justify-between  ml-[2px]">
+                        <Combobox.Input
+                            onChange={(event) => setQuery(event.target.value)}
+                            displayValue={() => query}
+                            placeholder="Search"
+                            className="border-none w-full"
+                        />
+                        <button
+                            onClick={submitSearch}
+                            className="bg-inherit border-none pr-3"
+                            type="submit"
+                        >
+                            <SearchIcon className="fill-sui-steel cursor-pointer" />
+                        </button>
+                    </div>
+                    <Combobox.Options
+                        static
+                        as="div"
+                        className="w-full h-[209px] md:h-[571px] overflow-auto mr-[-20px] pr-[20px]"
+                    >
+                        <VerticalList>
+                            {filteredModules.map((name, idx) => (
+                                <Combobox.Option key={name} value={name}>
+                                    {({ active }) => (
+                                        <div
+                                            key={idx}
+                                            className="min-w-full w-fit"
```

Comment:
Curious why we need this wrapper.

Reply:
This was required at one point with one iteration of the PR but looks like indeed this is no longer a requirement.